Modules/Office365DSC/DSCResources/MSFT_SPOHomeSite/MSFT_SPOHomeSite.psm1

```diff
@@ -0,0 +1,197 @@
+function Get-TargetResource
+{
+    [CmdletBinding()]
+    [OutputType([System.Collections.Hashtable])]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("Yes")]
+        [String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.String]
+        $Url,
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Getting configuration for hub site collection $Url"
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
+        -Platform PnP
+
+    $nullReturn = @{
+        $IsSingleInstance  = $IsSingleInstance
+        Ensure             = "Absent"
+        GlobalAdminAccount = $GlobalAdminAccount
+    }
+
+    try
+    {
+        Write-Verbose -Message "Getting current home site collection settings"
+        $homeSiteUrl = Get-SPOHomeSite
+        if ($null -eq $homeSiteUrl)
+        {
+            Write-Verbose -Message "There is no Home Site Collection set."
+            return $nullReturn
+        }
+        else
+        {
+            $result = @{
+                $IsSingleInstance  = $IsSingleInstance
+                $Url               = $homeSiteUrl
+                Ensure             = "Present"
+                GlobalAdminAccount = $GlobalAdminAccount
+            }
+            return $result
+        }
+    }
+    catch
+    {
+        Write-Verbose -Message "There was an error in the SPOHomeSite resource."
+    }
+    return $nullReturn
+}
+
+function Set-TargetResource
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateSet("Yes")]
+        [String]
+        $IsSingleInstance,
+
+        [Parameter()]
+        [System.String]
+        $Url,
+
+        [Parameter()]
+        [ValidateSet("Present", "Absent")]
+        [System.String]
+        $Ensure = "Present",
+
+        [Parameter(Mandatory = $true)]
+        [System.Management.Automation.PSCredential]
+        $GlobalAdminAccount
+    )
+
+    Write-Verbose -Message "Setting configuration for home site '$Url'"
+
+    Test-MSCloudLogin -O365Credential $GlobalAdminAccount `
```

Comment:
We are trying to move away from using SPO in favor of PnP. Use Get-PnPTenantSite -Url instead of Get-SPOSite to check for existence of the site collection